### ISSUE TITLE
Licence issue with Socksify gem

### ISSUE THREAD

**Issue by robbanp** [savonrb/httpi#205]
The socksify dependancy makes httpi not usable for commercial projects since it's using GPL licence. Or am I wrong? :)

**Comment by rogerleite** [comment]
Hi @robbanp 

It's a good question. I'm not a specialist on license subject, reading about it at Wikipedia, on section [Communicating and bundling with non-GPL programs](https://en.wikipedia.org/wiki/GNU_General_Public_License#Communicating_and_bundling_with_non-GPL_programs), it's said:

> An "aggregate" consists of a number of separate programs, distributed together on the same CD-ROM or other media. The GPL permits you to create and distribute an aggregate, even when the licenses of the other software are non-free or GPL-incompatible. The only condition is that you cannot release the aggregate under a license that prohibits users from exercising rights that each program's individual license would grant them.

I think we can consider HTTPI an "aggregate" and since it uses MIT as license, it's under the condition to not prohibit users from exercising rights on it.

This is my personal view, if you still in doubt or need an official answer, my suggestion is to contact a professional on subject.

If you know someone can help, feel free to copy here to help us.

**Comment by olleolleolle** [comment]
Seems like the socksify-ruby gemspec and code distribution has a [Ruby license](https://en.wikipedia.org/wiki/Ruby_License), as well.

**Comment by cptobvious** [comment]
@olleolleolle you're right about the [Ruby license](https://en.wikipedia.org/wiki/Ruby_License) in [socksify](https://github.com/astro/socksify-ruby). Did anyone bother to check the individual licenses/copying conditions of the files mentioned in https://github.com/astro/socksify-ruby/commit/42a51b7a95201d254e62c53322e9093cd3ce397e?
Imho it would be a cleaner usage scenario if socksify would be replaced.

**Comment by ioquatix** [comment]
I think we all agree to remove it: https://github.com/savonrb/httpi/issues/224 at least as a default dependency.
